### ISSUE TITLE
new auto train

### ISSUE THREAD

**Issue by marl0stanfield** [Sau1707/ModernBot#29]
Hey

The actual auto train is broken af because grepolis use severals request for train troops, and the request who is send by the bot its not same like the game.

so the only method for do an real auto train is simulate player click like :
1. switch to the city
2. open the barracks / docks
3. choose the troops
4. and recruit with the button ingame

if its not like this, the auto train is detected and you can be banned

**Comment by Sau1707** [comment]
I am aware of this, the problem is that the main Idea was to create a bot that work without the graphical interface interaction. I've decided to do this since make it more stable and robust.

I'm thinking of a solution to this that don't include the opening of the window but at the same time don't show up as a fake request.

**Comment by Sau1707** [comment]
I'm working to fix this, the code is almost ready for the buildings, the troops still has to be done

I've fixed the problem by using the Curator building when possible instead of the normal one (is still a fallback if the curator is not active), the same will happen for the training.

The code will automatically simulate the opening of the curator view when necessary and has a random delay to simulate the user action

**Comment by poko33** [comment]
Hello

Is there any code update?

**Comment by Sau1707** [comment]
Yeah...I stopped working on this since I'm very busy recently...

**Comment by poko33** [comment]
Is it possible to get the latest update?

**Comment by Sau1707** [comment]
Yeah not really...the code it's super buggy and could (and probably will) produce some unexpected behaviour that make it worst than the one it's there yet... maybe one day I'll find some time to fix it
